src/Curl/Curl.php

```diff
@@ -571,6 +571,46 @@ public function setVerbose($on = true)
         $this->setOpt(CURLOPT_VERBOSE, $on);
         return $this;
     }
+    
+    /**
+     * set the proxy
+     *
+     * @param   string $type      value can be CURLPROXY_HTTP(default),CURLPROXY_SOCKS4,CURLPROXY_SOCKS5,CURLPROXY_SOCKS4A and CURLPROXY_SOCKS5_HOSTNAME
+     * @param   int $port         default: 8080
+     * @param   string $host      default: 127.0.0.1
+     * @param   string $userpwd   default: Null, example: 'username.password', value are string
+     * @return  self
+     * @author  dugulingping
+     * @email   dugulingping@outlook.com
+     */
+    public function setProxy($type = 'CURLPROXY_HTTP',$port = 8080, $host = '127.0.0.1', $userpwd = null) : Curl{
```

Comment:
It is usual to put the host before the port.

What about username ?

Reply:
Now $host Before $port 

Now, The Username and Password at $userAndPasswd, They are merger, The format is: 'username.password'